### PR TITLE
Overview of CI in Contributing.md

### PR DESCRIPTION
**Pull request by BoD**

### CODE REVIEW

File: Contributing.md [hunk 1]
```diff
@@ -1,19 +1,16 @@
-Contributing to Apollo Android GraphQL 
-======================================
+# Contributing to Apollo Android GraphQL
```
Comment (martinbonnin) [inline Contributing.md:1]: Super Nit: I'd just go with `Apollo Android` for the project name (until it changes to something else maybe 👀 )

File: Releasing.md [hunk 2]
```diff
@@ -1,14 +0,0 @@
-# Before Releasing
-
-
-Run `./gradlew japicmp` and check what API may have changed. We ideally want 100% API and ABI compatibility but the project is still moving fast and it's not always possible.
```
Comment (martinbonnin) [inline Releasing.md:4]: 👍

File: Contributing.md [hunk 3]
```diff
@@ -61,3 +57,47 @@ and then push it.
...
+    - Slow gradle tests
+- `tests-no-gradle`
+    - All root tests but not the Gradle ones and not the "exotic" apple ones (tvos, watchos)
+    - All apiCheck
+- `tests-integration`
+    - All integration tests
+
+### On pushes to `dev-3.x` branch
+
+_(Will be replaced to "on pushes to `main` branch" after v3 is released.)_
```
Comment (martinbonnin) [inline Contributing.md:83]: I'm betting a pumkin pie that we're going to forget this and have docs out of sync with the code in a few weeks. 🙃

**Review by martinbonnin — APPROVED** [review]

File: Contributing.md (hunk 3 above)
Comment (BoD) [reply Contributing.md:83]: You are probably right 😛 Created [this issue](https://github.com/apollographql/apollo-android/issues/3525) so we don't forget.

**Pull request merged**